c2c: gather perf metrics from prometheus

msbutler opened the pull request:
c2c roachtest performance metrics are now gathered by a prom/grafana instance running locally on the roachprod cluster. This change allows us to gather and process any metrics exposed to the crdb prom endpoint. Specifically, we now gather: `capacity_used`, `replication_logical_bytes`, `replication_sst_bytes` at various points during the c2c roachtest, allowing us to measure:
- Initial Scan Throughput: initial scan size / initial scan duration
- Workload Throughput: data ingested during workload / workload duration
- Cutover Throughput: (data ingested between cutover time and cutover cmd) / (cutover process duration)

where the size of these operations can be measured as either physical replicated bytes, logical ingested bytes, or physical ingested bytes on the source cluster.

This patch also fixes a recent bug which mislabeled src cluster throughput as initial scan throughput.

Informs #89176

Release note: None

msbutler committed:
c2c: gather perf metrics from prometheus

c2c roachtest performance metrics are now gathered by a prom/grafana instance
running locally on the roachprod cluster. This change allows us to gather and
process any metrics exposed to the crdb prom endpoint. Specifically, we now
gather: `capacity_used`, `replication_logical_bytes`, `replication_sst_bytes`
at various points during the c2c roachtest, allowing us to measure:
- Initial Scan Throughput: initial scan size / initial scan duration
- Workload Throughput: data ingested during workload / workload duration
- Cutover Throughput: (data ingested between cutover time and cutover cmd) /
  (cutover process duration)

where the size of these operations can be measured as either physical
replicated bytes, logical ingested bytes, or physical ingested bytes on the
source cluster.

This patch also fixes a recent bug which mislabeled src cluster throughput as
initial scan throughput.

Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/97465)
<!-- Reviewable:end -->

msbutler committed:
c2c: refactor tenant ru limit removal in roachtests

This patch streamlines how we remove ru limiting for roachtests that use
tenants. For the c2c tests specifically, we know remove the limits on the dst
cluster tenant as soon as the replication stream begins.

Release note: None

msbutler committed:
c2c: add c2c/tpcc/warehouses=1000/duration=60/cutover=30 roachtest

Epic: none

Release note: none

stevendanna commented on pkg/cmd/roachtest/tests/cluster_to_cluster.go:
```diff
@@ -67,11 +70,68 @@ type c2cSetup struct {
+			if err != nil {
+				t.L().Errorf("Could not query prom %s", err.Error())
+			}
+			metricSnap[name] = sumOverLabel(point, stat.LabelName)
```
Is it possible to put this summation into the Query like you can in the grafana graph?

msbutler replied:
Sadly, not with this `collector.CollectPoint()` api. I'll leave a todo and file an issue.

msbutler commented:
TFTR! 

bors r=stevendanna
